The root cause of error is suppressed in KC 26 at building

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dependencies

### Describe the bug

If we have any error at KC's building phase, then the real root cause of  error is suppressed. Threrefore the KC's admin is not able to detect real root cause of error.

In KC 25 and less KC's admin is able to see real stacktrace. This bug is reproduced in version 26 only.

### Version

26.0.8

### Regression

- [x] The issue is a regression

### Expected behavior

The stacktrace looks like:
```
...
Caused by: java.lang.ClassNotFoundException: some.required.ClassName
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:525)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:473)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:523)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:473)
  ... 24 more
```

And root cause of error is clear. We missed dependency in `providers` directory.

### Actual behavior

The stacktrace looks like:
```
ERROR: Failed to run 'build' command.
Error details:
java.lang.NullPointerException
  at java.base/java.util.Objects.requireNonNull(Objects.java:233)
  at java.base/java.util.ImmutableCollections$List12.<init>(ImmutableCollections.java:563)
  at java.base/java.util.List.of(List.java:937)
  at io.quarkus.paths.OpenContainerPathTree.getRoots(OpenContainerPathTree.java:96)
  at io.quarkus.paths.SharedArchivePathTree$CallerOpenPathTree.getRoots(SharedArchivePathTree.java:142)
  at io.quarkus.bootstrap.classloading.PathTreeClassPathElement.toString(PathTreeClassPathElement.java:214)
  at java.base/java.util.Formatter$FormatSpecifier.printString(Formatter.java:3158)
  at java.base/java.util.Formatter$FormatSpecifier.print(Formatter.java:3036) 
```

The root cause is not clear.
Stacktrace of root cause is suppressed.

### How to Reproduce?

1. Add any extension, with a dependency to another jar, to `providers` directory
2. Exclude the dependency from `providers` directory
3. Run Keycloak building as `kc.bat|sh build --verbose`

### Anything else?

Existing issue with same error
https://github.com/keycloak/keycloak/issues/35742

Unfortunately the command:

```
./kc.sh -Dquarkus.launch.rebuild=true --help
```
also returns the masked stacktrace:
```
./kc.bat -Dquarkus.launch.rebuild=true --help
The DelayedHandler was closed before any children handlers were configured. Messages will be written to stderr.
2025-01-15 20:53:35,407 TRACE [java.io.serialization] (main) Builtin factory: null -> new: null
2025-01-15 20:53:35,838 TRACE [java.io.serialization] (main) Builtin factory: null -> new: null
2025-01-15 20:53:35,994 DEBUG [org.jboss.logging] (main) Logging Provider: org.jboss.logging.JBossLogManagerProvider
Exception in thread "main" java.lang.NullPointerException
  at java.base/java.util.Objects.requireNonNull(Objects.java:233)
  at java.base/java.util.ImmutableCollections$List12.<init>(ImmutableCollections.java:563)
  at java.base/java.util.List.of(List.java:937)                                                                                                                                                                     
  at io.quarkus.paths.OpenContainerPathTree.getRoots(OpenContainerPathTree.java:96)
  at io.quarkus.paths.SharedArchivePathTree$CallerOpenPathTree.getRoots(SharedArchivePathTree.java:142)
  at io.quarkus.bootstrap.classloading.PathTreeClassPathElement.toString(PathTreeClassPathElement.java:214)
```

Meanwhile KC 25 in the same case returns correct stacktrace:
```
...
Caused by: java.lang.ClassNotFoundException: some.required.ClassName
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:525)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:473)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:523)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:473)
  ... 24 more
```

> A build needs to run inbetween correct?

I'm assuming this is the case, and have updated the description.

> Caused by: java.lang.ClassNotFoundException: some.required.ClassName

You are saying in the decription that this can be any dependency, but I'm not sure that is accurate. I tested with a JDBC driver jar and didn't see the same behavior. 

Can you be more specific about is in the jar that is causing you problems - in particular does it have classes that would be used by quarkus bootstrapping?

I don't think this will be treated as a regression because we do document that you should not include classes that conflict with built-in jars in your provider jars.

Let me clarify,

1. "Add any extension with dependency to `providers` directory" means: there is some KC extension (let `super-extension.x.y.z.jar`) with dependency from `super-component-a.b.c.jar`.
2. "Exclude this dependency from `providers` directory" means: copy `super-extension.x.y.z.jar` to `providers` directory, actually we need to copy super-component-a.b.c.jar too, but for our case we need to miss it.
3. "Run Keycloak building as `kc.bat|sh build --verbose`" means: just prepare our extensions for KC, or we could just run KC (kc.bat|sh start --verbose`), afaik KC runs 'build' under the hood.

Expected behavior:
we see the stacktrace like this
```
...
Caused by: java.lang.ClassNotFoundException: some.required.ClassName
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:525)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:473)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:523)
  at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:473)
  ... 24 more
```
where `some.required.ClassName` is a class from our dependency, from `super-component-a.b.c.jar`.

Actual behavior:
we see the stacktrace like this
```
Exception in thread "main" java.lang.NullPointerException
  at java.base/java.util.Objects.requireNonNull(Objects.java:233)
  at java.base/java.util.ImmutableCollections$List12.<init>(ImmutableCollections.java:563)
  at java.base/java.util.List.of(List.java:937)                                                                                                                                                                     
  at io.quarkus.paths.OpenContainerPathTree.getRoots(OpenContainerPathTree.java:96)
  at io.quarkus.paths.SharedArchivePathTree$CallerOpenPathTree.getRoots(SharedArchivePathTree.java:142)
  at io.quarkus.bootstrap.classloading.PathTreeClassPathElement.toString(PathTreeClassPathElement.java:214)
```

Ther correct stackstrace is suppressed (hided).

--- a/pom.xml
+++ b/pom.xml
@@ -51,8 +51,8 @@
         <jboss.snapshots.repo.id>jboss-snapshots-repository</jboss.snapshots.repo.id>
         <jboss.snapshots.repo.url>https://s01.oss.sonatype.org/content/repositories/snapshots/</jboss.snapshots.repo.url>
 
-        <quarkus.version>3.15.3.1</quarkus.version>
-        <quarkus.build.version>3.15.3.1</quarkus.build.version>
+        <quarkus.version>3.15.4</quarkus.version>
+        <quarkus.build.version>3.15.4</quarkus.build.version>
 
         <project.build-time>${timestamp}</project.build-time>
 

